lib/cordial/products.rb

```diff
@@ -31,5 +31,32 @@ class Products
     def self.find(product_id:)
       client.get("/products/#{product_id}")
     end
+
+    # Create a new product.
+    #
+    # If the product already exists it will be updated.
+    # @example Usage.
+    #  Cordial::Products.create(
+    #    product_id: 1,
+    #    product_name: 'Test Product',
+    #    price: 99.99,
+    #    variants: [{
+    #      sku: '123',
+    #      attr: {
+    #        color: 'red',
+    #        size: 'L'
+    #      }
+    #    }]
+    #  )
+    def self.create(product_id:, product_name:, price:, variants:)
+      client.post('/products',
+                  headers: { 'Content-Type' => 'application/json' },
```

Comment:
Why do we need this?

Reply:
without this, the request with the nested hashes `variants: [{sku: '123', attr: {color: 'red', size: 'L'}}]`  returns the following error

```
=> {"error"=>true, "messages"=>{"validationErrors"=>{"variants"=>[{"sku"=>"sku is required"}]}}}
```

I guess that is a bug on the HTTParty Gem. I could fix the error adding this header and the `to_json` on the body params.

Reply:
I don't think this should be the job of the `create` method. I think it's reasonable to say that all requests are going to be of this type. Can we move this to the client?

Reply:
Updated :D